src/libs/ajax/AzureStorage.js

```diff
@@ -0,0 +1,152 @@
+  },
+
+  details: async (workspaceId = {}) => {
+    const res = await fetchWorkspaceManager(`workspaces/v1/${workspaceId}/resources?stewardship=CONTROLLED&limit=1000`,
+      _.merge(authOpts(), { signal })
+    )
+    const data = await res.json()
+    const storageAccount = _.find({ metadata: { resourceType: 'AZURE_STORAGE_ACCOUNT' } }, data.resources)
+
+    if (storageAccount === undefined) { // Internal users may have early workspaces with no storage account.
+      return {
+        location: undefined,
+        storageContainerName: undefined,
+        sas: { url: undefined, token: undefined }
+      }
+    } else {
+      const container = _.find(
+        {
+          metadata: { resourceType: 'AZURE_STORAGE_CONTAINER', controlledResourceMetadata: { accessScope: 'SHARED_ACCESS' } },
+          resourceAttributes: { azureStorageContainer: { storageAccountId: storageAccount.metadata.resourceId } }
+        },
+        data.resources
+      )
+      const sas = await AzureStorage(signal).sasToken(workspaceId, container.metadata.resourceId)
+
+      return {
+        location: storageAccount.resourceAttributes.azureStorage.region,
+        storageContainerName: container.resourceAttributes.azureStorageContainer.storageContainerName,
+        sas
+      }
+    }
+  },
+
+  listFiles: async (workspaceId, suffixFilter = '') => {
+    if (!workspaceId) {
+      return []
+    }
+
+    const { sas: { url, token } } = await AzureStorage(signal).details(workspaceId)
+    const azureContainerUrl = _.flow(
+      _.split('?'),
+      _.head,
+      Utils.append(`?restype=container&comp=list&${token}`),
+      _.join('')
+    )(url)
+
+    const res = await fetchOk(azureContainerUrl)
+    const text = await res.text()
+    const xml = new window.DOMParser().parseFromString(text, 'text/xml')
+    const blobs = _.map(
+      blob => ({
+        name: _.head(blob.getElementsByTagName('Name')).textContent,
+        lastModified: new Date(
+          _.head(blob.getElementsByTagName('Last-Modified')).textContent
+        ).getTime()
+      }),
+      xml.getElementsByTagName('Blob')
+    )
+
+    const filteredBlobs = _.filter(blob => _.endsWith(suffixFilter, blob.name), blobs)
+    return filteredBlobs
+  },
+
+  listNotebooks: async workspaceId => {
+    const notebooks = await AzureStorage(signal).listFiles(workspaceId, '.ipynb')
+    return _.map(notebook => ({ ...notebook, application: tools.Jupyter.label }), notebooks)
+  },
+
+  blob: (workspaceId, blobName) => {
+    const calhounPath = 'api/convert'
+
+    const getObject = async () => {
+      const azureStorageUrl = await getBlobUrl(workspaceId, blobName)
+
+      const res = await fetchOk(azureStorageUrl)
+      const text = await res.text()
+      return text
+    }
+
+    const getBlobUrl = async (workspaceId, blobName) => {
+      const { sas: { url, token } } = await AzureStorage(signal).details(workspaceId)
+      const encodedBlobName = encodeAzureAnalysisName(blobName)
```

Comment:
I realize this isn't new in this PR, but it doesn't make sense to use `encodeAzureAnalysisName` within a generic `blob` namespace. How would this be used with a non-analysis blob/file?